### DISCUSSION TITLE
API documentation, and list of available endpoints?

### DISCUSSION THREAD

**Discussion by jgrade-hdr** [netzero-labs/netzero#39] [General]
Hi all, 
I was planning to play around with the NetZero API this week and next, but I noticed that the API documentation does not seem to currently be available. When I go to https://docs.netzero.energy/, and then click the "Netzero API" link (https://www.netzero.energy/api) I get a Not Found error. I'm pretty sure that I had visited this page a week or 2 ago, and it included instructions for authorizing an API request with an authentication key, and some examples of curl commands to a couple of the endpoints. But I'm not seeing that anymore. Is this documentation available from another link?

Also, is there any more extensive documentation of all the available API endpoints, and their expected inputs and outputs? Does this project have a Swagger page, for example, in which I could test various API endpoints?

Thanks in advance for the help.

**Comment by zigam** [comment]
Thanks for the report, I fixed the doc link:
https://docs.netzero.energy/docs/tesla/API

The API is pretty basic right now, so that's all the documentation. We'll consider Swagger if the API expands.

**Comment by jgrade-hdr** [comment]
Thank you for the quick response. 

So are there any other public endpoints besides the Site Config?

**Reply by zigam** [reply]
Not currently. What were you hoping to accomplish?

**Comment by jgrade-hdr** [comment]
Sorry for late response. I'm just coming back to this after working on other things for the past month or so.  Mostly, I'm just trying to understand the full capability of your API, and evaluating whether it allows me to do things that are not easy to accomplish via Tesla's own API.

I was specifically curious about whether any of the IFTTT automation could be done via the API in such a way that the app would recognize it? Or vice-versa (be able to create an automation in the app, and then see that reflected in the API).

I also noticed that Tesla seems to have updated its own APIs to give access to a tariff API endpoint (I think this was something that was added earlier this year, as I don't remember seeing it in Tesla's API beforehand). I was wondering if this functionality exists in NetZero API, or if there are any plans to add it?

**Reply by zigam** [reply]
> I was specifically curious about whether any of the IFTTT automation could be done via the API in such a way that the app would recognize it? Or vice-versa (be able to create an automation in the app, and then see that reflected in the API).

I'm not sure I understand the question, but the API will give you the latest configuration and state, and this can be modified using either the API or the app (Netzero or Tesla). The configuration and state will also be reflected both in the API and apps.

> I also noticed that Tesla seems to have updated its own APIs to give access to a tariff API endpoint (I think this was something that was added earlier this year, as I don't remember seeing it in Tesla's API beforehand). I was wondering if this functionality exists in NetZero API, or if there are any plans to add it?

Not in the Netzero API, but there are app integrations for dynamic tariffs using the API:
https://docs.netzero.energy/docs/tariffs/Dynamic-Electricity-Pricing
